packages/jest-runtime/src/index.ts

```diff
@@ -792,9 +786,7 @@ export default class Runtime {
     const namedExports = new Set(exports);
 
     reexports.forEach(reexport => {
-      const resolved = this._resolveCjsModule(modulePath, reexport, {
-        conditions: this.esmConditions,
```

Comment:
this specifically was the bug - we passed ESM conditions when figuring out CJS `require`s.

I moved all conditions into `_resolveCjsModule` and `_resolveModule`, that way this typo is way harder to make (and we always passed the same options anyways, no need for it to be an argument at all)